### PR TITLE
Azure publicip plugin tests

### PR DESCRIPTION
**Pull request by wassoares**
This pull request contains the implementation of the tests from the public IP plug-in for the Azure cloud.

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    // observable executes without any error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeAssociatePublicIPAddressSuccessfully() {
+        // set up
+        Azure azure = PowerMockito.mock(Azure.class);
+        String instanceId = AzureGeneralUtil.defineInstanceId(AzureTestUtils.RESOURCE_NAME);
+
+        NetworkInterface networkInterface = Mockito.mock(NetworkInterface.class);
+        Observable<NetworkInterface> observable = Observable.defer(() -> {
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:58]: You should use a method that belongs to AzureTestUtils. 
[Link](https://github.com/fogbow/resource-allocation-service/blob/develop/src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/AzureTestUtils.java#L46)

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                .subscribeUpdateNetworkInterface(Mockito.eq(observable));
+    }
+
+    // test case: When calling the subscribeUpdateNetworkInterface method and the
+    // observable executes without any error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeUpdateNetworkInterfaceSuccessfully() {
+        // set up
+        Observable<NetworkInterface> observable = Observable.defer(() -> {
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:147]: Same as the previous issue.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        this.loggerAssert.assertEqualsInOrder(Level.ERROR, Messages.Error.ERROR_UPDATE_NIC_ASYNC_BEHAVIOUR);
+    }
+
+    // test case: When calling the subscribeDisassociateAndDeleteResources method
+    // and the observable executes without any error, it must verify than returns
+    // the right logs.
+    @Test
+    public void testSubscribeDisassociateAndDeleteResourcesSuccessfully() {
+        // set up
+        Completable completable = Completable.complete();
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:180]: You should use a method that belongs to AzureTestUtils.
[Link](https://github.com/fogbow/resource-allocation-service/blob/develop/src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/AzureTestUtils.java#L50)
Look at the whole code about this same issue.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java (hunk 1 above)
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:58]: The "Observable<Indexable>" type of the createSimpleObservableSuccess method in AzureTestUtils is not compatible with "Observable<NetworkInterface>".

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java (hunk 2 above)
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:147]: The "Observable<Indexable>" type of the createSimpleObservableSuccess method in AzureTestUtils is not compatible with "Observable<NetworkInterface>".

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java (hunk 3 above)
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java:180]: I'll fix It and the other occurrences.

**Pull request merged**